exporter/signalfxexporter/internal/translation/converter.go

```diff
@@ -88,13 +88,13 @@ func (c *MetricsConverter) MetricsToSignalFxV2(md pmetric.Metrics) []*sfxpb.Data
 			ilm := rm.ScopeMetrics().At(j)
 			for k := 0; k < ilm.Metrics().Len(); k++ {
 				dps := c.translator.FromMetric(ilm.Metrics().At(k), extraDimensions)
-				dps = c.translateAndFilter(dps)
-				sfxDataPoints = append(sfxDataPoints, dps...)
+				sfxInitialDataPoints = append(sfxInitialDataPoints, dps...)
```

Comment:
Hm... Looks like we need to empty `sfxInitialDataPoints` prior to entering this loop, otherwise we will be appending the same data multiple times to `sfxModifiedDataPoints` :)

Another suggestion: Since it's mostly only one scope metrics object generated by the scrapers, we can move `translateAndFilter` inside this loop. that should simplify things

Reply:
Good catch on emptying the `sfxInitialDataPoints` slice!

As far as moving the `translateAndFilter` call, having it outside the `scope` loop is essentially the purpose of this PR. It's to be able to properly account for the exporter's translations handling the removal of the `direction` dimension.

The question that I need answered then, is it guaranteed that both "directions" for metrics (e.g. `system.disk.io.read` and `system.disk.io.write`) will be in the **same** `ScopeMetric`? If yes, this PR can essentially be just updating tests to match proper metric format, but if not, we have to have the `translateAndFilter` call outside of the `ScopeMetric` loop.

Reply:
I believe I've answered my own question, but feel free to correct me if I've missed it. I ran the collector locally with the following flag: ```--feature-gates=-receiver.hostmetricsreceiver.emitMetricsWithDirectionAttribute,+receiver.hostmetricsreceiver.emitMetricsWithoutDirectionAttribute```, with the `network` scraper setup.

Also, [referencing the data model documentation](https://opentelemetry.io/docs/reference/specification/metrics/datamodel/#opentelemetry-protocol-data-model), the resource in `ResourceMetrics` is referring to the data source, being my local machine in my testing. `ScopeMetrics` is referring to `The instrumentation Scope (e.g., instrumentation library name, version)`. In my case, the scope name is `otelcol/hostmetricsreceiver/network` and `Version` is `latest`.

I think it's pretty safe to say that other hostmetrics will follow this same format, so metrics with different names, but from the same receiver and resource will still be in the same ScopeMetric. I believe this means we can make this PR simply test data format changes to follow proper metric specifications.

Reply:
> The question that I need answered then, is it guaranteed that both "directions" for metrics (e.g. system.disk.io.read and system.disk.io.write) will be in the same ScopeMetric?

Yes, all the scraper produce only one `ScopeMetrics` object within `ResourceMetrics`, so we can put it inside the loop iterating over ScopeMetrics with no issues (not inside the inner Metrics loop). It'll help to avoid emptying `sfxInitialDataPoints`.

Reply:
I've moved the call to be at the ScopeMetrics level.